### DISCUSSION TITLE
container-fluid not applying

### DISCUSSION THREAD

**Discussion by AdrielS30** [twbs/bootstrap#39988] [Q&A]
Hey all. Working on responsive design, React. 
When I start to shrink the viewport in devtools, my content begins to not take up the full width of the screen. When I delete my bootstrap import, the problem goes away (need it for other styling though). My component code begins with a main tag with the container-fluid class. The effect is being applied as seen in devtools, my content takes up 100% of the main tag width, as well as 100% of the <div id="root> tag and body tag from index.html, however all three of these are stopping very short of the full width of the screen. 
The body tag is being restricted by default bootstrap container settings, I think. My custom css is imported below bootstrap, ive tried setting body, html and root to 100% width, but nothing seems to work. Any ideas?

![Screen Shot 2024-05-15 at 1 18 18 PM (2)](https://github.com/twbs/bootstrap/assets/142935915/8240775d-3798-462a-a40d-adfa876b1345)

---

_Edited by maintainer to insert the image in the original post_

**Comment by gerardnico** [comment]
What we see is the mobile view and you seem to be at 100% of it. No?
Otherwise one of the parent is constrained in width.

Do you use a SSR framework to render react ?

**Reply by AdrielS30** [reply]
Thanks for your response. 

Yeah that's mobile view, and was assuming it should span 100% at that point.

And no I wasn't using an SSR, and definitely am going to start doing that.

I forget what the exact fix was at this point honestly. I think it was an overflow setting, and deleting most of my px margins and using rem's and %'s instead.
